tests/nominal/test_OneHotEncodingTransformer.py

```diff
@@ -264,3 +265,76 @@ def test_unseen_categories_encoded_as_all_zeroes(self, df_test, expected):
             actual=df_transformed,
             msg="unseen category rows not encoded as 0s",
         )
+
+
+class TestAdditionalOneHotEncodingTransformer:
+    """Additional tests for OneHotEncodingTransformer."""
+
+    @pytest.fixture
+    def sample_data(self):
+        return pd.DataFrame(
+            {
+                "color": ["red", "blue", "green", "blue", "red"],
+                "size": ["S", "M", "L", "S", "M"],
+            },
+        )
+
+    def test_not_fitted_error_transform(self, sample_data):
+        transformer = OneHotEncodingTransformer(columns=["color", "size"])
+        with pytest.raises(NotFittedError):
+            transformer.transform(sample_data)
+
+    def test_initialization_and_fitting(self, sample_data):
+        transformer = OneHotEncodingTransformer(columns=["color", "size"])
+        transformer.fit(sample_data)
+        assert hasattr(
+            transformer._encoder,
+            "categories_",
+        ), "The encoder should have 'categories_' attribute after fitting."
+
+    def test_transformation(self, sample_data):
```

Comment:
We already test outputs of transform in TestTransform.  Don't think this is adding anything new?  if it is then it should be in that test class with amore descriptive name.

Reply:
removed